index.html

```diff
@@ -2697,15 +2697,46 @@ <h3><dfn>Writing or pushing content</dfn></h3>
           |message:NDEFMessageSource| run these steps:
         </p>
         <ol class=algorithm id="create-web-nfc-message">
-          <li>
-            If the |message:NDEFMessageSource| parameter is not of type defined
-            by the <a>NDEFMessageSource</a> union, throw a {{TypeError}} and
-            abort these steps.
-          </li>
-          <li>
-            If the |message:NDEFMessageSource| parameter is of
-            <a>NDEFMessageInit</a> type, and |message|'s records
-            [= list/is empty =], throw a {{TypeError}} and abort these steps.
+          <li>Switch on |message:NDEFMessageSource|'s type:
+            <dl>
+              <dt>{{DOMString}}</dt>
+              <ul>
+                <li>
+                  Let |textRecord| be an <a>NDEFrecord</a> initialized with its
+                  |recordType| set to "`text`" and |data| set to |message|.
+                </li>
+                <li>
+                  Set |message|'s records to a list of records that contain only
+                  |textRecord|.
+                </li>
+              </ul>
+              <dt>{{BufferSource}}</dt>
+              <ul>
+                  <li>
+                    Let |mimeRecord| be an <a>NDEFrecord</a> initialized with its
+                    |recordType| set to "`mime`", |data| set to |message|, and
+                    |mediaType| set to "`application/octet-stream`".
+                  </li>
+                  <li>
+                    Set |message|'s records to a list of records that contain only
```

Comment:
im sure the infra spec has algos (you can refer to) on how to append to a list

Reply:
https://infra.spec.whatwg.org/#list-append

Reply:
Would this be more appropriate?

```
Let |records| be the list « |mimeRecord| ». 
Set |message|'s records to |records|.
```

Reply:
I think that is more clear, yes

Reply:
Fixed in https://github.com/w3c/web-nfc/pull/444/commits/1bcd82bcc77639f1775a5b2e6d9baafb018e7c98